components/action-mgt/org.wso2.carbon.identity.action.management/src/test/java/org/wso2/carbon/identity/action/management/ActionManagementServiceImplTest.java

```diff
@@ -0,0 +1,418 @@
+                action.getEndpoint().getAuthentication().getProperties().size());
+        Assert.assertEquals(creatingAction.getEndpoint().getAuthentication().getProperties().size(),
+                action.getEndpoint().getAuthentication().getPropertiesWithSecretReferences(action.getId()).size());
+        secretProperties = mapActionAuthPropertiesWithSecrets(action);
+        Assert.assertEquals(
+                action.getEndpoint().getAuthentication().getProperty(Authentication.Property.USERNAME).getValue(),
+                secretProperties.get(Authentication.Property.USERNAME.getName()));
+        Assert.assertEquals(
+                action.getEndpoint().getAuthentication().getProperty(Authentication.Property.PASSWORD).getValue(),
+                secretProperties.get(Authentication.Property.PASSWORD.getName()));
+    }
+
+    @Test(priority = 2, expectedExceptions = ActionMgtException.class,
+            expectedExceptionsMessageRegExp = "Unable to create an Action.")
+    public void testAddMaximumActionsPerType() throws ActionMgtException {
+
+        Action creatingAction = buildMockAction(
+                "PreIssueAccessToken",
+                "To configure PreIssueAccessToken",
+                "https://example.com",
+                buildMockBasicAuthentication("admin", "admin"));
+        action = serviceImpl.addAction(PRE_ISSUE_ACCESS_TOKEN, creatingAction,
+                tenantDomain);
+    }
+
+    @Test(priority = 3)
+    public void testGetActionsByActionType() throws ActionMgtException, SecretManagementException {
+
+        List<Action> actions = serviceImpl.getActionsByActionType(PRE_ISSUE_ACCESS_TOKEN, tenantDomain);
+        Assert.assertEquals(1, actions.size());
+        for (Action result: actions) {
+            Assert.assertEquals(action.getId(), result.getId());
+            Assert.assertEquals(action.getName(), result.getName());
+            Assert.assertEquals(action.getDescription(), result.getDescription());
+            Assert.assertEquals(action.getType().getActionType(), result.getType().getActionType());
+            Assert.assertEquals(action.getStatus(), result.getStatus());
+            Assert.assertEquals(action.getEndpoint().getUri(), result.getEndpoint().getUri());
+            Assert.assertEquals(action.getEndpoint().getAuthentication().getType(),
+                    result.getEndpoint().getAuthentication().getType());
+            secretProperties = mapActionAuthPropertiesWithSecrets(result);
+            Assert.assertEquals(
+                    result.getEndpoint().getAuthentication().getProperty(Authentication.Property.USERNAME).getValue(),
+                    secretProperties.get(Authentication.Property.USERNAME.getName()));
+            Assert.assertEquals(
+                    result.getEndpoint().getAuthentication().getProperty(Authentication.Property.PASSWORD).getValue(),
+                    secretProperties.get(Authentication.Property.PASSWORD.getName()));
```

Comment:
Verification -> Whether the properties are encrypted and replaced with secret reference
Let's use `action.getEndpoint().getAuthentication().getPropertiesWithSecretReferences()` to get the references for the secrets and verify the property values in the response as well.